### PR TITLE
chore: add checks for fonts and icons in request failed error handler from e2e base page.ts

### PR DESCRIPTION
**Pull request by lisli1**
#### Description of changes

- Added checks for 'fonts' and 'icons' in error handler in `page.ts` that was causing e2e test suites to fail when fonts failed to load

#### Pull request checklist

- [x] Addresses an existing issue: Temporary workaround for #923
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)
- [x] (UI changes only) Added screenshots/GIFs to description above **N/A**
- [x] (UI changes only) Verified usability with NVDA/JAWS **N/A**

### CODE REVIEW

File: src/tests/end-to-end/common/page.ts [hunk 1]
```diff
@@ -18,9 +18,10 @@ export class Page {
         underlyingPage.on('pageerror', error => {
             forceEventFailure(`'pageerror' (console.error) with stack: ${error.stack}`);
         });
-        underlyingPage.on('requestfailed', request => {
-            forceEventFailure(`'requestfailed' from '${request.url()}' with errorText: ${request.failure().errorText}`);
-        });
+        // Removing this error handler as a temporary workaround for #923
+        // underlyingPage.on('requestfailed', request => {
```
Comment (smoralesd) [inline src/tests/end-to-end/common/page.ts]: In general, we do not put comment code on master! 
If anything, I guess you should remove this altogether as git history will preserve this code anyway.

File: src/tests/end-to-end/common/page.ts [hunk 2]
```diff
@@ -18,9 +18,10 @@ export class Page {
...
+        //     forceTestFailure(`request failed - ${request.failure().errorText}, ${request.url()}`);
```
Comment (peterdur) [inline src/tests/end-to-end/common/page.ts]: Consider checking the URL (maybe for prefix `https://static2.sharepointonline.com`, or suffix `.woff2`) and not calling forceTestFailure if it matches.

Comment (lisli1) [reply src/tests/end-to-end/common/page.ts]: Updated to check for `fonts` and `icons` in the URL

File: src/tests/end-to-end/common/page.ts [hunk 3]
```diff
@@ -18,8 +19,12 @@ export class Page {
         underlyingPage.on('pageerror', error => {
             forceEventFailure(`'pageerror' (console.error) with stack: ${error.stack}`);
         });
+        // Modifying this error handler as a workaround for #923
```
Comment (Shobhit1) [inline src/tests/end-to-end/common/page.ts]: nit: I think it makes sense to not have this comment with issue reference here.

Comment (dbjorge) [reply src/tests/end-to-end/common/page.ts]: I'm not sure if you mean "you think it belongs next to the if statement instead" or "you don't think it's necessary at all". I agree with the former and strongly disagree with the latter.

**Comment by dbjorge** [comment]
@lisli1 Can you use the `test:` or `chore:` prefix for the PR/commit? `fix:` is for fixes to user-facing bugs (something that would motivate a new release, which we would want to include in release notes/release validation)

**Pull request merged**